kyo-data/shared/src/main/scala/kyo/Discriminator.scala

```diff
@@ -0,0 +1,9 @@
+package kyo
+
+/** A dummy class used to use to help the compiler distinguish between overloaded methods that have subtle type differences.
+  */
+sealed abstract class Discriminator {}
+
+object Discriminator:
+    private val cached  = new Discriminator {}
+    given Discriminator = cached
```

Comment:
a `given` should be a `lazy val` by default, right?
```suggestion
    given Discriminator = new Discriminator {}
```